src/client_side_request.cc

```diff
@@ -295,7 +295,7 @@ clientBeginRequest(const HttpRequestMethod& method, char const *url, CSCB * stre
     /* client stream setup */
     clientStreamInit(&http->client_stream, clientGetMoreData, clientReplyDetach,
                      clientReplyStatus, new clientReplyContext(http), streamcallback,
-                     streamdetach, streamdata, tempBuffer);
+                     streamdetach, std::move(streamdata), tempBuffer);
```

Comment:
Should not we fix clientStreamInit() API instead of expecting callers to remember to use an explicit std::move? The latter is a red flag because, buggy code notwithstanding, it is only useful in very limited circumstances. This code is not one of those places AFAICT.